### PR TITLE
KOGITO-3796 generic bulk list

### PR DESCRIPTION
**Pull request by AjayJagan**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by AjayJagan** [comment]
jenkins please retest this

**Comment by AjayJagan** [comment]
jenkins please retest this

**Review by nmirasch — APPROVED** [review]
Looks good and works fine, Thanks Ajay!

File: ui-packages/packages/management-console/src/components/Atoms/BulkList/BulkList.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+export enum OperationType {
+  ABORT = 'ABORT',
+  SKIP = 'SKIP',
+  RETRY = 'RETRY',
+  CANCEL = 'CANCEL'
+}
+
+export enum BulkListType {
+  PROCESS_INSTANCE = 'process_instnace',
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Atoms/BulkList/BulkList.tsx]:
```suggestion
  PROCESS_INSTANCE = 'process_instance',
```

File: ui-packages/packages/management-console/src/components/Atoms/BulkList/BulkList.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+              </strong>{' '}
+              {item.errorMessage && <span> - {item.errorMessage}</span>}
+            </TextListItem>
+          );
+        })}
+      </TextList>
+    );
+  };
+  return (
+    <div {...componentOuiaProps(ouiaId, 'bulk-list-instances', ouiaSafe)}>
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Atoms/BulkList/BulkList.tsx]: I would prefer the ouia type to be in sync with component name.
```suggestion
    <div {...componentOuiaProps(ouiaId, 'bulk-list', ouiaSafe)}>
```

**Review by jstastny-cz — COMMENTED** [review]
Just minor things.

File: ui-packages/packages/management-console/src/components/Atoms/BulkList/BulkList.tsx (hunk 1 above)
Comment (AjayJagan) [reply ui-packages/packages/management-console/src/components/Atoms/BulkList/BulkList.tsx]: done :+1:

File: ui-packages/packages/management-console/src/components/Atoms/BulkList/BulkList.tsx (hunk 2 above)
Comment (AjayJagan) [reply ui-packages/packages/management-console/src/components/Atoms/BulkList/BulkList.tsx]: sure, done :+1:

**Review by jstastny-cz — APPROVED** [review]
LGTM, thanks.

File: ui-packages/packages/management-console/src/components/Atoms/BulkList/BulkList.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  description: string;
+  errorMessage?: string;
+}
+
+const BulkList: React.FC<IOwnProps & OUIAProps> = ({
+  operationResult,
+  ouiaId,
+  ouiaSafe
+}) => {
+  const iterateItems = itemList => {
```
Comment (Sara4994) [inline ui-packages/packages/management-console/src/components/Atoms/BulkList/BulkList.tsx]: just a small nitpick can we add type here for the argument passed (itemList: BulkListItem[])

File: ui-packages/packages/management-console/src/components/Molecules/ProcessListToolbar/ProcessListToolbar.tsx [hunk 4]
```diff
@@ -363,14 +330,24 @@ const ProcessListToolbar: React.FC<IOwnProps & OUIAProps> = ({
       setIsAllChecked(false);
       setSelectedNumber(0);
       const copyOfInitData = { ...initData };
-      const copyOfSelectedInstances = { ...selectedInstances };
+      const copyOfSelectedInstances = [...selectedInstances];
       copyOfInitData.ProcessInstances.map(instance => {
-        delete copyOfSelectedInstances[instance.id];
+        //delete copyOfSelectedInstances[instance.id];
```
Comment (Sara4994) [inline ui-packages/packages/management-console/src/components/Molecules/ProcessListToolbar/ProcessListToolbar.tsx]: @AjayJagan please remove this line?

File: ui-packages/packages/management-console/src/components/Atoms/BulkList/BulkList.tsx (hunk 3 above)
Comment (AjayJagan) [reply ui-packages/packages/management-console/src/components/Atoms/BulkList/BulkList.tsx]: thanks :+1:  , done

File: ui-packages/packages/management-console/src/components/Molecules/ProcessListToolbar/ProcessListToolbar.tsx (hunk 4 above)
Comment (AjayJagan) [reply ui-packages/packages/management-console/src/components/Molecules/ProcessListToolbar/ProcessListToolbar.tsx]: yes , removed

**Comment by AjayJagan** [comment]
jenkins please retest this

**Comment by AjayJagan** [comment]
jenkins please retest this

**Comment by Sara4994** [comment]
Looks good to me now and good to merge after achieving green build.

**Review by Sara4994 — APPROVED** [review]

**Comment by jstastny-cz** [comment]
jenkins retest this

**Comment by AjayJagan** [comment]
jenkins retest this

**Pull request merged**